### PR TITLE
PR: allow JSON array as input

### PR DESCRIPTION
**Pull request by qistoph**
As described in https://github.com/sanack/node-jq/issues/363, node-jq currently does not allow JSON arrays as input.

This PR:
- adds test cases to indicate and test the issue
- fixes the input validation to allow arrays
- **possibly breaking** side affect: changes the undefined JSON input error slightly

```diff
-  'node-jq: invalid json object argument supplied: "undefined"'
+  'node-jq: invalid json argument supplied: "undefined"'
```

It's worth considering allow even more input, since any JSON object is effectively allowed by jq.

```bash
$ echo '1' | jq .
1
$ echo '9.99' | jq .
9.99
$ echo '"asdf"' | jq .
"asdf"
$ echo '["asdf","test"]' | jq .
[
  "asdf",
  "test"
]
$ echo 'null' | jq .
null
```

The only issue I can think of right now is with `undefined`:
```bash
$ echo 'undefined' | jq .                                                   
parse error: Invalid numeric literal at line 2, column 0                      
```

### CODE REVIEW

File: src/options.js [hunk 1]
```diff
@@ -48,7 +48,10 @@ export const preSpawnSchema = Joi.object({
           otherwise: path
         }).label('path')
     },
-    json: (schema) => Joi.object().allow('', null).required().label('json object'),
+    json: (schema) => Joi.alternatives().try(
```
Comment (davesnx) [inline src/options.js:51]: I'm not sure If I understand this change correctly. Isn't there any Joi.json instead?
I thought that `Joi.object()` would treat as JS Object so, same as `typeof [] === 'object'`

Comment (qistoph) [reply src/options.js:51]: Unfortunately that doesn't seem to be the case. If I send an array as input I get an error
**ValidationError: "json object" must be of type object**

You can checkout my test cases (99f8ae6b3383547d408f5e2aa282ab58f930a92a) to see what I'm trying to do.

Brief example:
```
const jq = require('node-jq')

obj = ["asdf","test"]

jq.run(".", JSON.stringify(obj), {'input': 'json'})
.then( out => {
    console.log("out:", out)
})
.catch( err => {
    console.error(err.message)
});
```

Comment (davesnx) [reply src/options.js:51]: Yep, totally get the why. I even wonder why do we even need Joi for the input tbh.

But this PR looks good.

**Review by davesnx — APPROVED** [review]

**Comment by davesnx** [comment]
I would add a breaking change since it does change the API surface. Maybe doable thought the commit, let me know if you need help with this, otherwise, I'm merging it.

**Comment by qistoph** [comment]
- Added 'BREAKING CHANGE' to the commit message.
- Rebased onto current master

**Pull request merged**